docs/root-filesystem.md

```diff
@@ -0,0 +1,60 @@
+# Root Filesystem
+
+firecracker-containerd requires a kernel and root filesystem to be specified for
+the Firecracker virtual machine where containers run.  This document specifies
+the hard requirements for that root filesystem, the development tools we built,
+and some thoughts around where we might land for a production recommendation.
+
+## Requirements
+
+* The firecracker-containerd agent, configured to start on boot
+* runc
```

Comment:
(This is more of a side-thought, up to you whether it's worth mentioning as a possibility in this doc)

I feel like we could get rid of the requirement for runc with only a moderate amount of effort. In theory, agent can just have runc statically compiled in and should be able to just use libcontainer rather than shell out to the separate runc binary. This would require some upstream changes, but nothing extraordinary.

This would be nice in that we can just drop the runc requirement entirely and only ask that users include the agent binary. It would also prevent any issues where agent and runc version get out of sync, should that possibility ever arise.

Reply:
I've added this as a question.

My inclination is to keep them separate so that it's easier to patch runc, but I don't have a strong opinion yet.